test/transactions/transactionBytes.js

```diff
@@ -15,50 +15,250 @@
+			const transactionBytes = getTransactionBytes(defaultTransaction);
+
+			(transactionBytes).should.be.eql(expectedBuffer);
+		});
+
+		it('should throw on type 0 with too much data', () => {
+			defaultTransaction.asset.data = '0123456789012345678901234567890123456789012345678901234567890123456789';
+			(getTransactionBytes.bind(null, defaultTransaction)).should.throw('Transaction asset data exceeds size of 64.');
+		});
+
+		it('should return Buffer of transaction with second signature', () => {
+			defaultTransaction.signSignature = defaultSecondSignature;
+			const expectedBuffer = Buffer.from('AKopAgBdA2qFjOifhESRdi64niv71QpKCg2mWOSyYoslsReuCQDOvKqNNBU96AMAAAAAAABhilSXUhLq2T34yIFlXGJVRLzo7XzN/m8IpC7s+xrevQUTB75QFLsFFhe694FdUPYhKecJGBkDYeXU3UeWVBsKsAxK0ZiLyiRddENWYKJ4v+a/L176i9qW2Sf6v4tPb8/cspU/arrKoRnWiAmHpV3qDmNUvINmBStF+iMUVSICDw==', 'base64');
+			const transactionBytes = getTransactionBytes(defaultTransaction);
+
+			(transactionBytes).should.be.eql(expectedBuffer);
+		});
 	});
 
-	it('should be a function', () => {
-		(getTransactionBytes).should.be.type('function');
+	describe('signature transaction, type 1', () => {
+		const signatureTransaction = {
+			type: 1,
+			amount: defaultNoAmount,
+			fee: 500000000,
+			recipientId: null,
+			senderPublicKey: defaultSenderPublicKey,
+			timestamp: defaultTimestamp,
+			asset: { signature: { publicKey: defaultSenderSecondPublicKey } },
+			signature: defaultSignature,
+			id: defaultTransactionId,
+		};
+
+		it('should create correct transactionBytes from signature transaction', () => {
+			const expectedBuffer = Buffer.from('AaopAgBdA2qFjOifhESRdi64niv71QpKCg2mWOSyYoslsReuCQAAAAAAAAAAAAAAAAAAAAAEAcisnyne2eHk1ba0MFHLJbIvJ8e3s1CSFh6FGUb4L2GKVJdSEurZPfjIgWVcYlVEvOjtfM3+bwikLuz7Gt69BRMHvlAUuwUWF7r3gV1Q9iEp5wkYGQNh5dTdR5ZUGwo=', 'base64');
+			const transactionBytes = getTransactionBytes(signatureTransaction);
+
+			(transactionBytes).should.be.eql(expectedBuffer);
+		});
 	});
 
-	it('should return Buffer of simply transaction and buffer most be 117 length', () => {
-		const transaction = {
-			type: 0,
-			amount: 1000,
-			recipientId: '58191285901858109L',
-			timestamp: 141738,
-			asset: {},
-			senderPublicKey: '5d036a858ce89f844491762eb89e2bfbd50a4a0a0da658e4b2628b25b117ae09',
-			signature: '618a54975212ead93df8c881655c625544bce8ed7ccdfe6f08a42eecfb1adebd051307be5014bb051617baf7815d50f62129e70918190361e5d4dd4796541b0a',
-			id: '13987348420913138422',
+	describe('delegate registration transaction, type 2', () => {
+		const delegateRegistrationTransaction = {
+			type: 2,
+			amount: defaultNoAmount,
+			fee: 2500000000,
+			recipientId: null,
+			senderPublicKey: defaultSenderPublicKey,
+			timestamp: defaultTimestamp,
+			asset: { delegate: { username: 'MyDelegateUsername' } },
```

Comment:
https://docs.lisk.io/docs/the-lisk-protocol-transactions lists `publicKey` as well as `username` for `asset.delegate`

Reply:
PublicKey will not be part of the delegate asset anymore:
https://github.com/LiskHQ/lisk-js/issues/296